pallets/ajuna-tournament/src/lib.rs

```diff
@@ -0,0 +1,571 @@
+		ValueQuery,
+	>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config<I>, I: 'static = ()> {
+		TournamentCreated { season_id: T::SeasonId, tournament_id: TournamentId },
+		TournamentStarted { season_id: T::SeasonId, tournament_id: TournamentId },
+		TournamentEnded { season_id: T::SeasonId, tournament_id: TournamentId },
+	}
+
+	#[pallet::error]
+	pub enum Error<T, I = ()> {
+		/// There's no active tournament for the selected season.
+		NoActiveTournamentForSeason,
+		/// There's already an active tournament for the selected season.
+		AnotherTournamentAlreadyActiveForSeason,
+		/// Cannot find tournament data for the selected season id and tournament id combination.
+		TournamentNotFound,
+		/// Cannot activate a tournament before its configured block start,
+		TournamentActivationTooEarly,
+		/// Cannot deactivate a tournament before its configured block end,
+		TournamentEndingTooEarly,
+		/// An error occurred trying to rank an entity,
+		FailedToRankEntity,
+		/// Tournament configuration is invalid.
+		InvalidTournamentConfig,
+	}
+
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// The account ID of the treasury.
+		pub fn global_treasury_account_id() -> AccountIdFor<T> {
+			T::PalletId::get().into_account_truncating()
+		}
+
+		/// The account ID of the sub-account for a given season_id/tournament_id.
+		pub fn tournament_treasury_account_id(
+			season_id: &T::SeasonId,
+			tournament_id: &TournamentId,
+		) -> T::AccountId {
+			let mut base = T::PalletId::get();
+			// TODO: Improve logic
+			base.0[0] = (*season_id).saturated_into::<u32>() as u8;
+			base.0[1] = *tournament_id as u8;
+			base.into_account_truncating()
+		}
+
+		fn ensure_valid_tournament(
+			config: &TournamentConfigFor<T, I>,
+		) -> Result<u16, DispatchError> {
+			ensure!(config.start < config.end, Error::<T, I>::InvalidTournamentConfig);
+
+			if let Some(initial_reward) = config.initial_reward {
+				ensure!(initial_reward > 0_u32.into(), Error::<T, I>::InvalidTournamentConfig);
+			}
+
+			if let Some(max_reward) = config.max_reward {
+				ensure!(max_reward > 0_u32.into(), Error::<T, I>::InvalidTournamentConfig);
+			}
+
+			if let Some(fee_perc) = config.take_fee_percentage {
+				ensure!(fee_perc <= 100, Error::<T, I>::InvalidTournamentConfig);
+			}
+
+			let reward_table_total_dist =
+				config.reward_table.iter().fold(0_u16, |a, b| a + (*b as u16));
+			ensure!(reward_table_total_dist <= 100, Error::<T, I>::InvalidTournamentConfig);
+
+			ensure!(
+				config.max_players > 0 && config.max_players <= MAX_PLAYERS,
+				Error::<T, I>::InvalidTournamentConfig
+			);
+
+			Ok(reward_table_total_dist)
+		}
+
+		fn try_get_current_tournament_id_for(
+			season_id: &T::SeasonId,
+		) -> Result<TournamentId, DispatchError> {
+			let maybe_id = ActiveTournaments::<T, I>::get(season_id);
+			ensure!(maybe_id.is_some(), Error::<T, I>::NoActiveTournamentForSeason);
+			Ok(maybe_id.unwrap())
+		}
```

Comment:
In practice this will probably not be overlooked, but I prefer not to see any `unwraps` in runtime code if it can be prevented. We can just return the following here:

```rust
ActiveTournaments::<T, I>::get(season_id).ok_or_else(|| Error::<T,I>::NoActiveTournamenForSeason)
```